mongo/collection.go

```diff
@@ -1474,6 +1734,40 @@ func (coll *Collection) findAndModify(ctx context.Context, op *operation.FindAnd
 	}
 }
 
+// mergeFindOneAndDeleteOptions combines the given FindOneAndDeleteOptions instances into a single
+// FindOneAndDeleteOptions in a last-one-wins fashion.
```

Comment:
Optional: see similar for explanation

```suggestion
// FindOneAndDeleteOptions in a last-property-wins fashion.
```